src/commons.ts

```diff
@@ -72,172 +70,26 @@ export default class Commons {
-    //         vscode.window.setStatusBarMessage("Sync : Updating In Progres... Please Wait.", 3000);
-    //     }
-    // });
-
-    Commons.configWatcher.onDidChange(async (uri: vscode.Uri) => {
-      const path: string = uri.path;
-
-      // check sync is locking
-      if (await lockfile.Check(this.en.FILE_SYNC_LOCK)) {
-        uploadStopped = false;
-      }
-
-      if (!uploadStopped) {
-        vscode.window.setStatusBarMessage("").dispose();
-        vscode.window.setStatusBarMessage(
-          localize("common.info.updating"),
-          3000
-        );
-        return false;
-      }
-
-      uploadStopped = false;
-      await lockfile.Lock(this.en.FILE_SYNC_LOCK);
-      const settings: ExtensionConfig = this.GetSettings();
-      const customSettings: CustomSettings = await this.GetCustomSettings();
-      if (customSettings == null) {
-        return;
-      }
-
-      let requiredFileChanged: boolean = false;
-      if (
-        customSettings.ignoreUploadFolders.indexOf("workspaceStorage") === -1
-      ) {
-        requiredFileChanged =
-          path.indexOf(this.en.FILE_SYNC_LOCK_NAME) === -1 &&
-          path.indexOf(".DS_Store") === -1 &&
-          path.indexOf(this.en.FILE_CUSTOMIZEDSETTINGS_NAME) === -1;
-      } else {
-        requiredFileChanged =
-          path.indexOf(this.en.FILE_SYNC_LOCK_NAME) === -1 &&
-          path.indexOf("workspaceStorage") === -1 &&
-          path.indexOf(".DS_Store") === -1 &&
-          path.indexOf(this.en.FILE_CUSTOMIZEDSETTINGS_NAME) === -1;
-      }
-
-      console.log("Sync : File Change Detected On : " + path);
-
-      if (requiredFileChanged) {
-        if (settings.autoUpload) {
-          if (
-            customSettings.ignoreUploadFolders.indexOf("workspaceStorage") > -1
-          ) {
-            const fileType: string = path.substring(
-              path.lastIndexOf("."),
-              path.length
-            );
-            if (fileType.indexOf("json") === -1) {
-              console.log(
-                "Sync : Cannot Initiate Auto-upload on This File (Not JSON)."
-              );
-              uploadStopped = true;
-              return;
-            }
-          }
-
-          console.log("Sync : Initiating Auto-upload For File : " + path);
-          this.InitiateAutoUpload(path)
-            .then(isDone => {
-              uploadStopped = isDone;
-              return lockfile.Unlock(this.en.FILE_SYNC_LOCK);
-            })
-            .catch(() => {
-              uploadStopped = true;
-              return lockfile.Unlock(this.en.FILE_SYNC_LOCK);
-            });
-        }
-      } else {
-        uploadStopped = true;
-        await lockfile.Unlock(this.en.FILE_SYNC_LOCK);
-      }
-    });
+  ) {
+    this.InitializeAutoUpload();
   }
 
-  public async InitiateAutoUpload(path: string): Promise<boolean> {
-    vscode.window.setStatusBarMessage("").dispose();
-    vscode.window.setStatusBarMessage(

```

Comment:
We need to show user if autoUpload process start processing. Have you moved this status or please add this where the `autoUpload` process is started.

Reply:
Should I leave the 5 second delay, or just start uploading right away?

Reply:
You can leave 5 second delay.

Reply:
Ok, I have done that.